substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/configure/ReflectionConfigurationParser.java

```diff
@@ -80,14 +82,29 @@ private void parseClass(Map<String, Object> data) {
 
         Object classObject = data.get("name");
         String className = asString(classObject, "name");
+        ConfigurationPredicate predicate = ConfigurationPredicate.DEFAULT_CONFIGURATION_PREDICATE;
+        Object predicateData = data.get("predicate");
+        if (predicateData != null) {
+            Map<String, Object> predicateObject = asMap(predicateData, "Attribute 'predicate' must be an object");
+            Object predicateType = predicateObject.get("whenTypeReachable");
+            if (predicateType instanceof String) {
+                String preticateTypeName = (String) predicateType;
+                TypeResult<T> result = delegate.resolveTypeResult(predicate, preticateTypeName);
+                if (!result.isPresent()) {
+                    handleError("Could not resolve " + preticateTypeName + " as predicate for reflection configuration.", result.getException());
+                    return;
+                }
+                predicate = ConfigurationPredicate.create(delegate.getTypeName(result.get()));
+            }
```

Comment:
I don't think we should ignore when `predicateType` is some other value than a string.

Reply:
Agree, we need to report a warning here. The warning allows us to extend this feature further, for example to arrays or a whole new language. 

Fixed.